tests/report_test.sh

```diff
@@ -379,6 +379,24 @@ function test_save_data_to()
   output=$(save_data_to '/this/is/An/InvaLid/Path')
   ret="$?"
   assert_equals_helper "We expect an invalid path error" "$LINENO" "$ret" 1
+
+  # Try to use a valid directory path
+  output=$(save_data_to './')
+  ret="$?"
+
+  assert_equals_helper "We expect a valid path" "$LINENO" "$ret" 0
```

Comment:
Here you should also check that the file was created with the expected name, and that the correct message was displayed.